src/cryptonote_core/blockchain.cpp

```diff
@@ -2194,7 +2194,11 @@ bool Blockchain::find_blockchain_supplement(const std::list<crypto::hash>& qbloc
 
   bool result = find_blockchain_supplement(qblock_ids, resp.m_block_ids, resp.start_height, resp.total_height);
   if (result)
-    resp.cumulative_difficulty = m_db->get_block_cumulative_difficulty(resp.total_height - 1);
+  {
+    cryptonote::difficulty_type wide_cumulative_difficulty = m_db->get_block_cumulative_difficulty(resp.total_height - 1);
+    resp.cumulative_difficulty = wide_cumulative_difficulty.convert_to<uint64_t>();
```

Comment:
If we're uncertain whether converting to a smaller uint will drop the higher bits or not, we should use bitshift operator (<< 64 >> 64) or bitmask 0xffffffffffffffff to be sure.